tools/run_tests/xds_test_driver/framework/infrastructure/gcp/compute.py

```diff
@@ -0,0 +1,336 @@
+            'name': name,
+            'url_map': url_map.url,
+        })
+
+    def delete_target_http_proxy(self, name):
+        self._delete_resource(self.api.targetHttpProxies(),
+                              targetHttpProxy=name)
+
+    def create_forwarding_rule(
+            self,
+            name: str,
+            src_port: int,
+            target_proxy: GcpResource,
+            network_url: str,
+    ) -> GcpResource:
+        return self._insert_resource(
+            self.api.globalForwardingRules(),
+            {
+                'name': name,
+                'loadBalancingScheme':
+                    'INTERNAL_SELF_MANAGED',  # Traffic Director
+                'portRange': src_port,
+                'IPAddress': '0.0.0.0',
+                'network': network_url,
+                'target': target_proxy.url,
+            })
+
+    def delete_forwarding_rule(self, name):
+        self._delete_resource(self.api.globalForwardingRules(),
+                              forwardingRule=name)
+
+    @staticmethod
+    def _network_endpoint_group_not_ready(neg):
+        return not neg or neg.get('size', 0) == 0
+
+    def wait_for_network_endpoint_group(self, name, zone):
+
+        @retrying.retry(retry_on_result=self._network_endpoint_group_not_ready,
+                        stop_max_delay=60 * 1000,
+                        wait_fixed=2 * 1000)
+        def _wait_for_network_endpoint_group_ready():
+            try:
+                neg = self.get_network_endpoint_group(name, zone)
+                logger.debug(
+                    'Waiting for endpoints: NEG %s in zone %s, '
+                    'current count %s', neg['name'], zone, neg.get('size'))
+            except googleapiclient.errors.HttpError as error:
+                # noinspection PyProtectedMember
+                reason = error._get_reason()
+                logger.debug('Retrying NEG load, got %s, details %s',
+                             error.resp.status, reason)
+                raise
+            return neg
+
+        network_endpoint_group = _wait_for_network_endpoint_group_ready()
+        # @todo(sergiitk): dataclass
+        return self.ZonalGcpResource(network_endpoint_group['name'],
+                                     network_endpoint_group['selfLink'], zone)
+
+    def get_network_endpoint_group(self, name, zone):
+        neg = self.api.networkEndpointGroups().get(project=self.project,
+                                                   networkEndpointGroup=name,
+                                                   zone=zone).execute()
+        # @todo(sergiitk): dataclass
+        return neg
+
+    def wait_for_backends_healthy_status(
+            self,
+            backend_service,
+            backends,
+            timeout_sec=_WAIT_FOR_BACKEND_SEC,
+            wait_sec=4,
+    ):
+        pending = set(backends)
+
+        @retrying.retry(retry_on_result=lambda result: not result,
```

Comment:
The reason is I started with `retyring`, then wasn't happy with how it difficult it made provide timeout/wait variables. I don't like having all these nested methods.
Then at the very last moment I found out that `tenacity` is a thing and it superseded `retrying`, which is now is abandoned.

> It originates from a fork of retrying which is sadly no longer maintained. Tenacity isn’t api compatible with retrying but adds significant new functionality and fixes a number of longstanding bugs.
> — https://tenacity.readthedocs.io/en/latest/index.html

Would it be ok with you if I replaced retrying with tenacity everywhere in the following iteration.

Reply:
That makes sense. Maybe a top-level TODO in the readme about replacing retrying with tenacity? But yeah, this can wait - I hadn't heard of either library before, and in looking at tenacity's docs, I didn't see immediately that it was a fork until I realized the usage examples for both libraries were remarkably similar.